### ISSUE TITLE
`ecs.Entity` pool ergonomics

### ISSUE THREAD

**Issue by ImVexed** [mlange-42/arche#305]
Back again!

I managed to get a panic out of `arche` and wanted to get some advice on how I should be holding onto entities.

I have a simple tracking system that looks for entities with a `Target` component. It takes that target component out, which contains an `*ecs.Entity` value. It then gets the `Position` of the target to update its own `Velocity` towards the target. 

While doing those checks I do an initial sanity check to assert that the entity is alive. And I managed to get:
```
panic: runtime error: index out of range [1326464512] with length 3070

goroutine 14047 [running]:
github.com/mlange-42/arche/ecs.(*entityPool).Alive(...)
        /home/co/go/pkg/mod/github.com/mlange-42/arche@v0.8.0/ecs/pool.go:74
github.com/mlange-42/arche/ecs.(*World).Alive(...)
        /home/co/go/pkg/mod/github.com/mlange-42/arche@v0.8.0/ecs/world.go:430
gameserver/game/systems.(*TrackingSystem).Update(0xc00065a6e0, 0xc0000fe9d8)
        /home/co/code/towerdefensegame/gameserver/server/game/systems/tracking.go:53 +0x3f0
github.com/mlange-42/arche-model/model.(*Systems).updateSystems(0xc0000fe900)
```

My code looks like:
```
	query := s.filter.Query(w)
	for query.Next() {
		// Get the speed and target of the projectile
		id, pos, velocity, speed, target := query.Get()

		// If the target has no entity ignore it
		if target.Entity == nil {
			continue
		}

                // If the entity is dead, handle retargeting...
		if !w.Alive(*target.Entity) {
			target.Entity = nil
			continue
		}
```

I'm assuming that the entity is getting deleted after being killed, but another entity is still holding onto it as a target and somehow, it's getting recycled into non-zero memory? I've already footguned myself a few times here holding onto entities after they've been deleted thinking the value should be safe, but it seems not?

**Comment by mlange-42** [comment]
Entities are not intended for use with pointers. Rather, they are very lightweight and should always be "copied" (simply by assignment). That's also the reason why you find (almost?) no entity pointers in the API.

Instead of `nil`, use the zero entity value (uninitialized or `target.entity = Entity{}`). To check for the zero value, use `target.entity.IsZero()`.

I think this should solve the issue and other such footguns. I should probably document this more prominently.

**Comment by ImVexed** [comment]
That makes sense, thanks! Will update if I manage to get a panic there again, it's not easy to reproduce.

**Comment by mlange-42** [comment]
FYI: I updated the docs in #306 and released v0.8.1.

**Comment by mlange-42** [comment]
Closing as resolved
